src/network/networkd-dhcp4.c

```diff
@@ -339,20 +339,19 @@ static int dhcp4_request_route_auto(
 
         } else {
                 if (in4_addr_is_null(gw)) {
```

Comment:
Please reduce the indentation here, and keep the debugging log with some modification:
```c
} else if (in4_addr_is_null(&gw)) {
        log_link_debug(link, "DHCP: requested route destination "IPV4_ADDRESS_FMT_STR"/%u is not in the assigned network "
                       IPV4_ADDRESS_FMT_STR"/%u, but no gateway is specified, using 'link' scope.",
                       ...);
        route->scope = RT_SCOPE_LINK;
        route->gw_family = AF_UNSPEC;
        route->gw = IN_ADDR_NULL;
        route->prefsrc.in = address;
} else {
        ...;
}
```

Reply:
Thanks! Updated.